lib/JSON/Validator.pm

```diff
@@ -875,10 +864,11 @@ sub _validate_type_object {
       : $self->_validate($data, $path, $schema->{then} // {});
   }
 
-  my $coerce = $self->{coerce}{defaults};
-  while (my ($k, $r) = each %{$schema->{properties}}) {
+  my %rules;
+  for my $k (keys %{$schema->{properties}}) {
```

Comment:
Why did you change from `while` to `for` here?

Reply:
I explained that in the comment for commit  46c2b23.